src/indexing.jl

```diff
@@ -139,6 +138,7 @@ You can also work with it directly, say you wanted to get all checkpoints files
 2: https://github.com/JuliaData/DataFrames.jl
 """
 function index_checkpoint_files(dir::AbstractPath)
+    isdir(dir) || throw(ArgumentError(dir, "Need an existing directory."))
```

Comment:
this isn't a valid `ArgumentError` call 
```jl
julia> dir = "foo"
"foo"

julia> throw(ArgumentError(dir, "Need an existing directory."))
ERROR: MethodError: no method matching ArgumentError(::String, ::String)
Closest candidates are:
  ArgumentError(::AbstractString) at boot.jl:317
  ArgumentError(::Any) at boot.jl:317
Stacktrace:
 [1] top-level scope
   @ REPL[5]:1
```

we should add a test for this change, maybe something like `@test_throws ArgumentError index_checkpoint_files("nonexistent_dir")`

Reply:
ping @bencottier @aisopous